server/src/main/java/com/objectcomputing/checkins/services/memberprofile/currentuser/CurrentUserServicesImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.objectcomputing.checkins.services.memberprofile.currentuser;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileServicesImpl;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+public class CurrentUserServicesImpl implements CurrentUserServices {
+
+    @Inject
+    MemberProfileServicesImpl memberProfileServicesImpl;
+
+    @Inject
+    MemberProfileRepository memberProfileRepo;
+
+    @Override
+    public MemberProfile findOrSaveUser(@NotNull String name, @NotNull String workEmail) {
+
+        Optional<MemberProfile> userProfile = memberProfileRepo.findByWorkEmail(workEmail);
+        if (userProfile.isPresent()) {
+            return userProfile.get();
+        }
+
+        return memberProfileServicesImpl.saveProfile(new MemberProfile(name,"",null,
```

Comment:
Thoughts behind this? I think I get what you are doing here, but is this what we want?

Reply:
I want to make sure that a member profile is in place for every user that logs into the app. Or is that not what you were asking?

Reply:
when an existing user enters the application, we'll find memberProfile using email address and return it.

when a new user enters the application, we'll create a record and return it. Sadly, name, email address and image_url are the only usable fields from Auth token. The users could go to profile page to complete their profile.

Reply:
That answers it. Thanks.